sapi/cli/php_cli_server.c

```diff
@@ -2053,6 +2060,12 @@ static int php_cli_server_begin_send_static(php_cli_server *server, php_cli_serv
 	int fd;
 	int status = 200;
 
+	if (client->request.request_method == PHP_HTTP_DELETE
+		|| client->request.request_method == PHP_HTTP_PUT
+		|| client->request.request_method == PHP_HTTP_PATCH) {
+		return php_cli_server_send_error_page(server, client, 405);
```

Comment:
When the server responds with 405, it is supposed to generate an `Allow` header field in the response; is this already implemented? Adding a test case for this scenario seems to be a good idea.